components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/ThemeManagerHostObject.java

```diff
@@ -380,6 +382,21 @@ private static void deployCustomTheme(FileHostObject themeFile, String tenantDom
         }
     }
 
+    private static String validateFilename(String filename, String intendedDir)
+            throws java.io.IOException {
+
+        File f = new File(filename);
+        String canonicalPath = f.getCanonicalPath();
+
+        File iD = new File(intendedDir);
+        String canonicalID = iD.getCanonicalPath();
```

Comment:
better to have more meaningful names for the 'f' and 'iD' variables. It is hard to understand what those names mean without checking the variable declaration. Also, require corresponding meaningful names for 'canonicalPath' and 'canonicalID'